publication-rest/src/main/java/no/unit/nva/publication/PublicationResponseFactory.java

```diff
@@ -0,0 +1,72 @@
+package no.unit.nva.publication;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.util.Optional;
+import java.util.Set;
+import no.unit.nva.PublicationMapper;
+import no.unit.nva.api.PublicationResponse;
+import no.unit.nva.api.PublicationResponseElevatedUser;
+import no.unit.nva.clients.IdentityServiceClient;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifact;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifactList;
+import no.unit.nva.model.associatedartifacts.file.HiddenFile;
+import no.unit.nva.publication.permission.strategy.PublicationPermissionStrategy;
+import nva.commons.apigateway.RequestInfo;
+
+public final class PublicationResponseFactory {
+
+    private PublicationResponseFactory() {
+    }
+
+    public static PublicationResponse create(Publication publication, RequestInfo requestInfo,
+                                             IdentityServiceClient identityServiceClient) {
+        var userStrategy = getPublicationPermissionStrategy(requestInfo, publication, identityServiceClient);
+
+        if (userStrategy.isPresent() && hasAuthenticatedAccessOnPublication(userStrategy.get())) {
+            return createAuthenticatedResponse(userStrategy.get(), publication);
+        } else {
+            return createPublicResponse(publication);
+        }
+    }
+
+    private static boolean hasAuthenticatedAccessOnPublication(PublicationPermissionStrategy userStrategy) {
+        return userStrategy.allowsAction(PublicationOperation.UPDATE);
+    }
+
+    private static PublicationResponse createPublicResponse(Publication publication) {
+        var publicationResponse = PublicationMapper.convertValue(publication, PublicationResponse.class);
+        publicationResponse.setAllowedOperations(Set.of());
```

Comment:
Not sure why `Set.of()` is preferred over `emptySet()` since the latter makes it extremely clear what the intention is.